### PR TITLE
Cleanup MaaS entity and agent tokens during gate trap

### PR DESCRIPTION
**Pull request by npawelek**
Ensure that gating properly deletes the agent token and the entity (if possible) in the test-ansible-functional.sh gate trap. This will run whether rpc-maas gating tests pass or fail to ensure that resources are cleaned up.

Issue: INSIGHTS-89

### CODE REVIEW

**Comment by npawelek** [comment]
recheck_trusty_liberty_deploy
recheck_trusty_mitaka_deploy

**Comment by npawelek** [comment]
recheck_xenial_pike_deploy

File: playbooks/library/raxmon.py [hunk 1]
```diff
@@ -150,6 +155,18 @@ def create_agent_token(module, conn, entity):
...
 
+def delete_agent_token(module, conn, entity):
+    agent_tokens = _get_agent_tokens(conn, entity)
+    for token in agent_tokens:
+        try:
+            conn.delete_agent_token(token)
+        except Exception as e:
+            msg = "Deleting agent token for %s failed. Reason:\n" % token.label
+            msg += str(e.message)
+            module.fail_json(msg=msg)
```
[the remark this answers is not in the document]
Comment (npawelek) [reply playbooks/library/raxmon.py]: @tonytan4ever There should only be a single entity/agent token created per job. The maas-agent-setup.yml playbook will fail if more than one entity exists also.

**Comment by hughsaunders** [comment]
recheck_all failures may have been caused by Jenkins outage RE-2210.

**Comment by npawelek** [comment]
@tonytan4ever This was added to the `gate_job_exit_tasks` function in this PR. You mentioned `gate_exit_job`, but I assume you mean the same thing?

**Comment by tonytan4ever** [comment]
> @tonytan4ever This was added to the `gate_job_exit_tasks` function in this PR. You mentioned `gate_exit_job`, but I assume you mean the same thing? This should run regardless of success/fail of the test-ansible-functional.sh script. If it fails prior to this, the entity will automatically be cleaned up via the server deletion process and the agent token will not exist yet.

Yes, I meant `gate_job_exit_tasks`

**Pull request merged**